koku/koku/middleware.py

```diff
@@ -341,11 +341,20 @@ def process_request(self, request):  # noqa: C901
 
         account = json_rh_auth.get("identity", {}).get("account_number")
         org_id = json_rh_auth.get("identity", {}).get("org_id")
-        user = json_rh_auth.get("identity", {}).get("user", {})
-        username = user.get("username")
-        email = user.get("email")
-        is_admin = user.get("is_org_admin")
+        token_type = str(json_rh_auth.get("identity", {}).get("type", "user")).lower()
+        user = None
+        email = None
+        is_admin = False
         req_id = None
+        if token_type == "user":
+            user = json_rh_auth.get("identity", {}).get("user", {})
+            username = user.get("username")
+            email = user.get("email")
+            is_admin = user.get("is_org_admin")
+        else:
```

Comment:
Should we always assume that a token_type that is not `user` will always be `service-account`?

Reply:
If we do and it doesn't contain a service account object then we will end up with a 401, but if you would prefer an `elseif` I can do that.

Reply:
Eh, I'm fine with a 401. I just wanted to verify that is handled correctly when we receive neither `user` nor `service-account`.